src/ImpactX.cpp

```diff
@@ -336,4 +336,35 @@ namespace impactx {
             }, element_variant);
         }
     }
+
+    void
+    ImpactX::track_covariance_map ()
+    {
+        // TODO: move whole body out in separate file
+
+        // TODO: init done?
+        amrex::ParmParse const pp_dist = amrex::ParmParse("beam");
+        auto ref = initialization::read_reference_particle(pp_dist);
+        auto dist = initialization::read_distribution(pp_dist);
+        auto cm = impactx::initialization::create_covariance_matrix(dist);
+
+        // TODO: output of init state?
+
+        // loop over all beamline elements
+        for (auto & element_variant : m_lattice)
+        {
+            std::visit([&ref, &cm](auto&& element){
+                // push reference particle in global coordinates
+                {
+                    BL_PROFILE("impactx::Push::RefPart");
+                    element(ref);
+                }
+
+                // push Covariance Matrix
+                cm = cm * element.transport_map(ref);
```

Comment:
Correct would be: `R_T * cm * R`

Reply:
Transpose is on the right:  R * cm * R_T

Reply:
Thanks! Dang, we had a 50% chance to get it right by guessing :D

Reply:
fixed